homeassistant/components/overkiz/executor.py

```diff
@@ -41,9 +41,13 @@ def device(self) -> Device:
         """Return Overkiz device linked to this entity."""
         return self.coordinator.data[self.device_url]
 
-    def linked_device(self, index: int) -> Device:
+    def linked_device(self, index: int) -> Device | None:
         """Return Overkiz device sharing the same base url."""
-        return self.coordinator.data[f"{self.base_device_url}#{index}"]
+
+        if (device_key := f"{self.base_device_url}#{index}") in self.coordinator.data:
+            return self.coordinator.data[device_key]
+
+        return None
```

Comment:
```suggestion
        device_key = f"{self.base_device_url}#{index}")
        return self.coordinator.data.get(device_key)
```
You could simplify this if you wanted to. (Just a suggestion)